### ISSUE TITLE
ENH: Option for pull and build to delete the existing image if present

### ISSUE THREAD

**Issue by jakirkham** [docker/docker#18072]
Instead of leaving a dangling image around or potentially purging common layers to early, it would be nice if there was an option to cleanly replace an image when pulling or building and tagging so there will not be a dangling image left behind.

**Comment by thaJeztah** [comment]
Thanks for suggesting! I think the new content addressability storage now implements garbage collecting, and automatically cleans up unused image layers (PR: https://github.com/docker/docker/pull/17924)

ping @tonistiigi am I right here?

**Comment by tonistiigi** [comment]
@thaJeztah Yes, layers are automatically deleted but that does not mean any UI changes like that, as will not automatically remove images or change tagging rules.

**Comment by thaJeztah** [comment]
@tonistiigi If I do `docker pull ubuntu`, and an updated Ubuntu image is pulled, the old image is untagged; will that image automatically be garbage collected (if it's no longer in use by a container?), because I think that's the use case here.

**Comment by tonistiigi** [comment]
@thaJeztah No it won't be. You will have a dangling image like you had before. Only thing different is that this new image and old image may share some layers and if you delete the old one, the layers that are not referenced by any other image get deleted as well.

**Comment by thaJeztah** [comment]
@tonistiigi ah, thanks for clearing that up :+1: might be a nice addition (think there were plans a while ago to automatically garbage collect dangling images as well)

**Comment by LK4D4** [comment]
Related to garbage collection.

**Comment by thaJeztah** [comment]
Linking to the data management proposal; https://github.com/docker/docker/issues/22871
